iocore/utils/PropertyBlock.h

```diff
@@ -0,0 +1,254 @@
+#pragma once
+#ifndef _PropertyBlock_h
+#define _PropertyBlock_h
+
+#include "stdint.h"
+#include <cstddef>
+#include <atomic>
+#include <cstring>
+#include <vector>
+#include <type_traits>
+
+#include "ts/ink_assert.h"
+
+/// runtime defined static container
+/** The size of this structure is actually zero, so it will not change the size of your derived class.
+ * But new and delete are overriden to use allocate enough bytes of the derived type + properties added
```

Comment:
I'm not going to hit all of them, but do a pass of proof reading on the comments. It "pey"s. :-)